helpers/request.helper.js

```diff
@@ -0,0 +1,60 @@
+import axios from 'axios';
+import { config } from '../config';
+
+const headers = {
+  JSON: 'application/json',
+};
+
+export class RequestHelper {
+  static getRequestFormData(data) {
+    const formData = new FormData();
+    data.forEach(({ key, value }) => formData.append(key, value));
+
+    return formData;
+  }
+}
+
+export const requestService = (service, requestParams, requestConfig) => {
+  const token = localStorage.getItem('accessToken');
+
+  return axios.post(config.SERVER_URL + service, requestParams, {
+    headers: {
+      Authorization: 'Bearer ' + token,
+      'Content-Type': headers.JSON,
+    },
+    ...requestConfig,
+  });
+};
+
+export const parseAxiosResponse = (response) => response.data;
+
+export const parseErrorResponse = (error) => {
+  if (error && error.response && error.response.data) {
+    return error.response.data.msg;
+  }
+};
+
+export const requestFileContent = async (service, requestParams) => {
+  const token = localStorage.getItem('accessToken');
+
+  const response = await fetch(config.SERVER_URL + service, {
+    method: 'POST',
+    body: JSON.stringify(requestParams),
+    headers: {
+      Authorization: 'Bearer ' + token,
+      'Content-Type': headers.JSON,
+    },
+  }).then((res) => {
+    const header = res.headers.get('content-type');
+    const isJsonResponse = header.substring(0, 16) === headers.JSON;
+
+    return new Promise(async (resolve, reject) => {
+      if (isJsonResponse) {
+        return reject(await res.json());
+      }
+      resolve(await res.blob());
+    });
+  });
```

Comment:
Avoid to mix `then` with `await` 
```suggestion
  const response = await fetch(config.SERVER_URL + service, {
    method: 'POST',
    body: JSON.stringify(requestParams),
    headers: {
      Authorization: 'Bearer ' + token,
      'Content-Type': headers.JSON,
    },
  })
  const header = response.headers.get('content-type');
  const isJsonResponse = header.substring(0, 16) === headers.JSON;
  const finalResponse = await (isJsonResponse ? res.json() : res.blob())
```

Reply:
I need to check on that, I'm pretty sure that came as my last resort ;-)

Reply:
Yep, I couldnt get it to work with await (your solution doesn't seem to do the job either)
https://stackoverflow.com/questions/54385676/whats-wrong-with-awaiting-a-promise-chain/54387912#54387912
So it says in short that this mixture is not elegant and makes the code more error-prone. Fine. I will stick to then this time on parent level function and leave the still ugly async promise 
```js
return fetch(config.SERVER_URL + service, {
    method: 'POST',
    body: JSON.stringify(requestParams),
    headers: {
      Authorization: 'Bearer ' + token,
      'Content-Type': headers.JSON,
    },
  }).then((res) => {
    const header = res.headers.get('content-type');
    const isJsonResponse = header.substring(0, 16) === headers.JSON;

    return new Promise(async (resolve, reject) => {
      if (isJsonResponse) {
        return reject(await res.json());
      }
      resolve(await res.blob());
    });
  });
```